File: web-client/src/app/interfaces/car-interfaces/car.ts
import { CarType } from "../../enums/car-type";
import { CarDocuments } from "./car-documents";
import { CarStatus } from "./car-status";
import { SeatModel } from "./seat";
import { TripModel } from "../trip-interfaces/trip";

export interface CarModel {
    id: number;
    modelName: string;
    registNum: string;
    carType: CarType;
    seats: SeatModel[];
    carDocuments: CarDocuments[]
    carStatus: CarStatus;
    trips?: TripModel[],
    createdAt?: Date
}

File: TripsWebApp.DAL/Entities/CarEntities/Car.cs
using BlaBlaCar.DAL.Entities.TripEntities;

namespace BlaBlaCar.DAL.Entities.CarEntities
{
    public class Car : BaseEntity
    {
        public string ModelName { get; set; }
        public string RegistrationNumber { get; set; }
        public CarType CarType { get; set; }
        public Guid UserId { get; set; }
        public Status CarStatus { get; set; }
        public ICollection<CarDocuments> CarDocuments { get; set; }
        public ApplicationUser User { get; set; }
        public ICollection<Seat> Seats { get; set; }
        public ICollection<Trip>? Trips { get; set; }
    }
}
